### ISSUE TITLE
An unhandled exception: Could not inject EventInvoker method on type

### ISSUE THREAD

**Issue by tmt242001** [Fody/PropertyChanged#333]
[PropertyChangedIssue.zip](https://github.com/Fody/PropertyChanged/files/1900843/PropertyChangedIssue.zip)

Try building PropertyChangedIssue Project. you will get the below error:

**This works in Version 2.2.9.**

Error		Fody: An unhandled exception occurred:
Exception:
Failed to execute weaver C:\Users\ElvinT\.nuget\packages\propertychanged.fody\2.5.3\netclassicweaver\PropertyChanged.Fody.dll
Type:
System.Exception
StackTrace:
   at InnerWeaver.ExecuteWeavers()
   at InnerWeaver.Execute()
Source:
FodyIsolated
TargetSite:
Void ExecuteWeavers()
Could not inject EventInvoker method on type 'PropertyChangedIssue.Configuration'. It is possible you are inheriting from a base class and have not correctly set 'EventInvokerNames' or you are using a explicit PropertyChanged event and the event field is not visible to this instance. Either correct 'EventInvokerNames' or implement your own EventInvoker on this class. If you want to suppress this place a [DoNotNotifyAttribute] on PropertyChangedIssue.Configuration.
Type:
Fody.WeavingException
StackTrace:
   at ModuleWeaver.InjectMethod(TypeDefinition targetType, InvokerTypes& invokerType)
   at ModuleWeaver.AddOnPropertyChangedMethod(TypeDefinition targetType)
   at ModuleWeaver.FindMethodsForNodes()
   at ModuleWeaver.Execute()
   at InnerWeaver.ExecuteWeavers()
Source:
PropertyChanged.Fody
TargetSite:
Mono.Cecil.MethodDefinition InjectMethod(Mono.Cecil.TypeDefinition, InvokerTypes ByRef)
	PropertyChangedIssue		1

**Comment by jianliulin** [comment]
netcore sdk 2.1.104  the same iss

**Comment by Alexrgs** [comment]
Check if you have the package FodyPackaging on your project.

**Comment by ltrzesniewski** [comment]
Sorry, that's my fault, due to #319... I'll submit a PR to fix that.

First: that has nothing to do with FodyPackaging, as that package is meant to be used by Fody addins _only_.

The issue here is that your `ObservableObject` class doesn't expose a proper event invoker method, so PropertyChanged.Fody injects one, but it's named `<>OnPropertyChanged`, which is not included in the default event invoker names.

So, what's the fix? Just add this to the `ObservableObject` class:

```C#
protected void NotifyPropertyChanged([CallerMemberName] string propertyName)
    => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
```

That's it. If you want some optimisations, you can add this one instead (or in addition to the previous one):

```C#
protected void NotifyPropertyChanged(PropertyChangedEventArgs eventArgs)
    => PropertyChanged?.Invoke(this, eventArgs);
```

This is the preferred approach. An alternative would be to add `<>OnPropertyChanged` to your `EventInvokerNames`:

```XML
<Weavers>
  <PropertyChanged EventInvokerNames="&lt;&gt;OnPropertyChanged, OnPropertyChanged, NotifyPropertyChanged"/>
</Weavers>
```

But that's brittle. Adding a proper method is the better approach.

Note that the error message actually hints at what you need to do 😉 

> or implement your own EventInvoker on this class

**Comment by ltrzesniewski** [comment]
Ok, #334 makes the example project compile just fine, thanks for the repro!

**Comment by tmt242001** [comment]
Issue is fixed in 2.5.4
